src/Repository.php

```diff
@@ -179,7 +180,7 @@ public function updateModel(array $data, Model $model)
      *
      * @param int $perPage
      * @param array $columns
-     * @return \Illuminate\Pagination\AbstractPaginator
+     * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
```

Comment:
The problem is that this isn't always a LengthAwarePaginator. It could be a simple paginator :(